pkg/retryablehttp/httpclient.go

```diff
@@ -0,0 +1,83 @@
+package retryablehttp
+
+import (
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+const backOffMaxDuration = 3 * time.Second
+
+type RetryableRoundTripper struct {
+	Client *RetryableHTTPClient
+	once   sync.Once
+}
+
+// RoundTripper is an interface representing the ability to execute a single HTTP transaction, obtaining the Response for a given Request.
+// A RoundTripper must be safe for concurrent use by multiple goroutines.
+var _ http.RoundTripper = (*RetryableRoundTripper)(nil)
+
+func (rt *RetryableRoundTripper) init() {
+	if rt.Client == nil {
+		rt.Client = NewClient()
+	}
+}
+
+// RoundTrip executes a single HTTP transaction, but does not attempt to read the response, modify it, or close the body.
+func (rt *RetryableRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	rt.once.Do(rt.init)
+	return rt.Client.Do(req)
+}
+
+type RetryableHTTPClient struct {
+	internalClient http.Client
```

Comment:
I want to make a retryable client that has its own Transport (for TLS). One way to do this is to use `NewClient` and then set `internalClient`. I am doing that here https://github.com/openfga/openfga/pull/2/files#diff-0b4e82b4c716948a5e14e84e6695d4fd755a5ea47aa0742573dce8fdd2b20bb4R218. In order to do this here I think I will need this to be public. Is that okay?

Reply:
I added a new constructor